i18n_generate.mll

```diff
@@ -118,50 +132,87 @@ let print_header fmt ?primary_module ~default_language () =
   in
   Format.pp_print_string fmt @@
   default_lang ^
-   "let%server _language_ = " ^ server_language_reference ^ "\n\
-   let%server get_language () = Eliom_reference.Volatile.get _language_\n\
-   let%server set_language language = \n\
-   Eliom_reference.Volatile.set _language_ language\n\
-   \n\
-   let%client _language_ = " ^ client_language_reference ^ "\n\
-   let%client get_language () = !_language_\n\
-   let%client set_language language = _language_ := language\n\
+  "let%server _language_ = " ^ server_language_reference ^ "\n\
+  let%server get_language () = Eliom_reference.Volatile.get _language_\n\
+  let%server set_language language = \n\
+  Eliom_reference.Volatile.set _language_ language\n\
+  \n\
+  let%client _language_ = " ^ client_language_reference ^ "\n\
+  let%client get_language () = !_language_\n\
+  let%client set_language language = _language_ := language\n\
+  \n\
+  let%shared txt = Eliom_content.Html.F.txt\n\
+  "
+
+let print_header fmt ?primary_module ~default_language () =
+  let default_lang =
+    match primary_module with
+    | None -> "let default_language = " ^ default_language ^ "\n"
+    | Some module_name -> ""
+  in
+  Format.pp_print_string fmt @@
+  default_lang ^
+  "let _language_ = default_language \n\
+   let get_language () = your_function_to_getting_language \n\
+   let set_language language = \n\
+   your_function_to_setting_language\n\
    \n\
-   let%shared txt = Eliom_content.Html.F.txt\n\
-"
+   let txt = txt \n\
+  "
 
 (** Print the function [string_of_language] returning the string representation of a
     value o type t. The string representation is simply the value as a string. For
     example, the string representation of [Us] is ["Us"]
 *)
-let print_string_of_language fmt ~variants ~strings =
+let print_string_of_language_eliom fmt ~variants ~strings =
   Format.pp_print_string fmt "let%shared string_of_language = function \n" ;
   List.iter2 (fun v s -> Format.fprintf fmt "| %s -> %S" v s)
     variants strings ;
   Format.pp_print_string fmt "\n"
 
+let print_string_of_language fmt ~variants ~strings =
+  Format.pp_print_string fmt "let string_of_language = function \n" ;
+  List.iter2 (fun v s -> Format.fprintf fmt "| %s -> %S" v s)
+    variants strings ;
+  Format.pp_print_string fmt "\n"
+
+
 (** Print the function [language_of_string] returning the value of type t which
     corresponds to the given string. The exception [Unknown_language] is raised with
     the given string if the language doesn't exist.
 *)
-let print_language_of_string fmt ~variants ~strings =
+let print_language_of_string_eliom fmt ~variants ~strings =
   Format.pp_print_string fmt "let%shared language_of_string = function\n" ;
   List.iter2 (fun v s -> Format.fprintf fmt "| %S -> %s" s v)
     variants strings ;
   Format.pp_print_string fmt "| s -> raise (Unknown_language s)\n"
 
-let print_guess_language_of_string fmt =
+let print_language_of_string fmt ~variants ~strings =
```

Comment:
these functions are similar enough that I would put them under a helper function: 
```
  let helper_print_language_of_string ~eliom fmt ~variants ~strings =
    Format.pp_print_string fmt
    (if eliom then "let language_of_string = function\n"
     else "let%shared language_of_string = function\n") ;
    List.iter2 (fun v s -> Format.fprintf fmt "| %S -> %s" s v)
      variants strings ;
    Format.pp_print_string fmt "| s -> raise (Unknown_language s)\n"

   let print_language_of_string_eliom  = helper_print_language_of_string ~eliom:true
   let print_language_of_string = helper_print_language_of_string ~eliom:false

```

Reply:
I will do that!